bevy_lint/src/lints/nursery/camera_modification_in_fixed_update.rs

```diff
@@ -0,0 +1,194 @@
+//! `Camera` component.
+//!
+//! # Motivation
+//!
+//! Modifying camera components in the `FixedUpdate` schedule can cause jittery or inconsistent
+//! visuals because `FixedUpdate` runs at a fixed timestep mainly for physics and deterministic
+//! logic. The camera should be updated in the last variable timestep before `FixedUpdate` which is
+//! the `Update` schedule.
+//!
+//! # Known Issues
+//!
+//! - The lint only detects systems that explicitly use the `With<Camera>` query filter.
+//!
+//! # Example
+//!
+//! ```rust
+//! use bevy::prelude::*;
+//!
+//! fn move_camera(mut query: Query<&mut Transform, With<Camera>>) {
+//!     // ...
+//! }
+//!
+//! fn main() {
+//!     App::new()
+//!         .add_systems(FixedUpdate, move_camera);
+//! }
+//! ```
+//!
+//! Use instead:
+//!
+//! ```rust
+//! use bevy::prelude::*;
+//!
+//! fn move_camera(mut query: Query<&mut Transform, With<Camera>>) {
+//!     // ...
+//! }
+//!
+//! fn main() {
+//!     App::new()
+//!         .add_systems(Update, move_camera);
+//! }
+//! ```
+
+use clippy_utils::{diagnostics::span_lint_and_help, sym, ty::match_type};
+use rustc_hir::{ExprKind, QPath, def::Res};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::{Adt, GenericArgKind, TyKind};
+use rustc_span::Symbol;
+
+use crate::{declare_bevy_lint, declare_bevy_lint_pass, utils::hir_parse::MethodCall};
+
+declare_bevy_lint! {
+    pub CAMERA_MODIFICATION_IN_FIXED_UPDATE,
+    super::Nursery,
+    "Camera modification in FixedUpdate schedule",
+}
+
+declare_bevy_lint_pass! {
+    pub CameraModificationInFixedUpdate => [CAMERA_MODIFICATION_IN_FIXED_UPDATE],
+    @default = {
+        add_systems: Symbol = sym!(add_systems),
+    },
+}
+
+impl<'tcx> LateLintPass<'tcx> for CameraModificationInFixedUpdate {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx rustc_hir::Expr<'tcx>) {
+        if expr.span.in_external_macro(cx.tcx.sess.source_map()) {
+            return;
+        }
+
+        let Some(MethodCall {
+            method_path, args, ..
+        }) = MethodCall::try_from(cx, expr)
+        else {
+            return;
+        };
+
+        // Match calls to `App::add_systems(schedule, systems)`
+        if method_path.ident.name != self.add_systems {
+            return;
+        }
+
+        // First argument must be the `ScheduleLabel`
+        let Some(first_arg) = args.first() else {
+            return;
+        };
+
+        let schedule_ty = cx.typeck_results().expr_ty(first_arg).peel_refs();
+
+        // Skip if the schedule is not `FixedUpdate`
+        if !match_type(cx, schedule_ty, &crate::paths::FIXED_UPDATE) {
+            return;
+        }
+
+        let Some(systems) = args.get(1) else {
+            return;
+        };
+
+        // Collect all added system expressions
+        let system_exprs = if let ExprKind::Tup(inner) = systems.kind {
+            inner.iter().collect()
```

Comment:
Clever way of converting an `&[Expr]` to an `[&Expr]`! It took me a second to figure out why you were calling this and not `Vec::from(inner)` ^^